File: Lab1/Domain/smart_thermostat.py
from Lab1.Domain.smart_device import SmartDevice

class SmartThermostat(SmartDevice):
    def __init__(self, name, temperature, temperature_unit):
        self.__name = name
        self.__temperature = temperature
        self.__temperature_unit = temperature_unit
        self.__state = False
        self.__connected = False

    def turn_on(self):
        self.__state = True

    def turn_off(self):
        self.__state = False

    def is_on(self):
        return self.__state

    def get_temperature(self):
        return self.__temperature

    def update_temperature(self, temperature):
        self.__temperature = temperature

    def connect(self):
        print(f"{self.__name} connecting...")
        self.__connected = True
        print(f"{self.__name} connected")

    def disconnect(self):
        print(f"{self.__name} disconnecting...")
        self.__connected = False
        print(f"{self.__name} disconnected")

File: Lab1/src/Domain/Classes/SmartThermostat.java
package Domain.Classes;

import Domain.Interfaces.ISmartDevice;
import Domain.Interfaces.ISmartThermostat;

public class SmartThermostat implements ISmartThermostat {
    private boolean state;
    private boolean connectionState;
    private int temperature;

    public SmartThermostat() {
        this.state = false;
        this.connectionState = false;
        this.temperature = 20;
    }

    public void setTemperature(int temperature) {
        this.temperature = temperature;
    }

    public int getTemperature() {
        return this.temperature;
    }

    @Override
    public boolean getState() {
        return this.state;
    }

    @Override
    public boolean getConnectionState() {
        return this.connectionState;
    }

    @Override
    public void on() {
        System.out.println("thermostat turning on...");
        this.state = true;
    }

    @Override
    public void off() {
        System.out.println("thermostat turning off...");
        this.state = false;
    }

    @Override
    public void connect() {
        System.out.println("thermostat connecting...");
        this.connectionState = true;
    }

    @Override
    public void disconnect() {
        System.out.println("thermostat disconnecting...");
        this.connectionState = false;
    }
}
